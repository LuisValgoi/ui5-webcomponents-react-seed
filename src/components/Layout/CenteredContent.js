import React from 'react'

const style = {
  centered: {
    maxWidth: '970px',
    padding: '1rem',
    margin: '0 auto'
  }
}

export default function CenteredContent({ children }) {
  return (
    <div data-testid='centeredContent' style={style.centered}>
      {children}
    </div>
  )
}
