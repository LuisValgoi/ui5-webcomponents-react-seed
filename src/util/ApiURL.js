export default {
  GET_USER_LOGGED: '/v1/user/logged',
  GET_TODO_LIST: '/v1/todo/all',
  GET_TODO_BY_ID: '/v1/todo/:id'
};
