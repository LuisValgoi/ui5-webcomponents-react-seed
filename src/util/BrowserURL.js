export default {
  HOME: '/',
  TODO_LIST: '/todo/list',
  TODO_DETAIL: '/todo/:id/detail',
  NOT_FOUND: '/notFound',
  BUGGY: '/buggy',
  ANY: '/*'
};
