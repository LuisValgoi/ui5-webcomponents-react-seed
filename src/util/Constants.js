export default {
  REACT_QUERY: {
    CODES: {
      SUCCESS: 'success',
      LOADING: 'loading',
      ERROR: 'error',
    },
    KEYS: {
      RQ_GET_USER_LOGGED: 'RQ_GET_USER_LOGGED',
      RQ_GET_TODO_LIST: 'RQ_GET_TODO_LIST',
      GET_TODO_BY_ID: 'GET_TODO_BY_ID',
    },
  },
  SEED: {
    SELECTED_THEME: 'SELECTED_THEME',
  },
};
