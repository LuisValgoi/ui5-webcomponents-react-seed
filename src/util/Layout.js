import React from 'react';
import { Text } from '@ui5/webcomponents-react/lib/Text';

export const TextWrapper = ({ children, isOnHeader, style }) => {
  return (
    <Text
      tooltip={children}
      wrapping
      className={isOnHeader && 'text-header-color'}
      style={{ ...style, lineHeight: '20px' }}
    >
      {children}
    </Text>
  );
};
