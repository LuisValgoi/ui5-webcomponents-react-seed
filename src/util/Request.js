import axios from 'axios';

const Request = axios.create({
  baseURL: process.env.REACT_APP_API_URL
});

export default Request;
