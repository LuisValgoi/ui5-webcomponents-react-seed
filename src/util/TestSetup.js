import React from 'react';

import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { rest } from 'msw';
import { setupServer } from 'msw/node';

import BrowserProvider from './browser/BrowserProvider';

const render = (ui, { route = BrowserProvider.getUrl('HOME'), ...renderOptions } = {}) => {
  const WrapperProvider = ({ ...props }) => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return (
      <BrowserRouter>
        <Router history={history}>{props.children}</Router>
      </BrowserRouter>
    );
  };

  return rtlRender(ui, { wrapper: WrapperProvider, ...renderOptions });
};

const serverCustom = (apiUrl, data) => {
  return setupServer(
    rest.get(`*${apiUrl}`, (req, res, ctx) => {
      return res(ctx.json(data));
    }),
  );
};

export * from '@testing-library/react';

export { render, serverCustom };
