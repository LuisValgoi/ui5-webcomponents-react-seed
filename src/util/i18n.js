import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import LanguageDetector from 'i18next-browser-languagedetector';

import translationEN from '../locales/en/translation.json';
import translationPT from '../locales/pt/translation.json';

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    debug: true,
    keySeparator: false,
    interpolation: {
      escapeValue: false,
    },
    resources: {
      en: {
        translation: translationEN,
      },
      pt: {
        translation: translationPT,
      },
    },
    react: {
      useSuspense: false,
    },
  });

export default i18n;
