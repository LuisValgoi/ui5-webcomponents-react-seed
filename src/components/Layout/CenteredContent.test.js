import React from 'react';

import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../util/TestSetup';

import CenteredContent from './CenteredContent';

describe('Validator.js Test Suite', () => {
  beforeEach(() => {
    render(<CenteredContent />);
  });

  test('should render', () => {
    const content = screen.getByTestId('centered-content');

    expect(content).toBeInTheDocument();
  });

  test('should have a minWidth of 50% when desktop', () => {
    const content = screen.getByTestId('centered-content');

    expect(content).toBeInTheDocument();
    expect(content).toHaveStyle('min-width:50%');
  });
});
