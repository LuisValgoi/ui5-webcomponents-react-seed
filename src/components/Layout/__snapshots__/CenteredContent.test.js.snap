// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="centeredContent"
    style="max-width: 970px; padding: 1rem; margin: 0px auto;"
  />
</DocumentFragment>
`;
